docs/dev/developer-guides/transactions/blocks.md

```diff
@@ -1,50 +1,49 @@
+L2 blocks, or just "blocks", are simply the blocks created on L2, that is on the zkSync network. They are not included on the Ethereum chain. An L1 rollup block, which we call "batch", is a set of 
+consecutive (L2) blocks, it contains all the transactions, and in the same order, from the first block in the batch to the last block in the 
+batch.
 
-Blocks are strictly ordered (each new block has a connection to its parent block), and transactions within blocks are strictly ordered to keep track of the history of transactions.
-Unless there are exceptional circumstances, all network participants agree with the precise number and history of blocks at any particular time and attempt to combine the pending live transaction requests into the upcoming block.
+L1 batches, as the name suggests, are submitted to Ethereum. The main reason to have these different notions is that a block can 
+contain a minimal number of transactions, and thus be processed quickly, while in a batch we would like to include many transactions, to make the cost of interaction with L1 spread among many transactions.
 
-Every node adds the newly created block to the end of its blockchain after it has been assembled by one of the network's validators, and a new validator is chosen to build the following block.
 
-## Block numbers and time
+##  Block and Batch numbers
 
-On zkSync, when users send transactions, those transactions are included in L2 blocks. These L2 blocks are sealed every X seconds (X is not decided yet, but it will surely be in the 1–5 range). Typically, the user needs to wait at most X seconds to get the receipt and be informed that his transaction is included in some block.
-In L1 batches, we send some information about L2 blocks to the L1 contract (known as a commitment); later, we generate proof for the blocks and send them to the L1 contract. The L1 contract then verifies that the given proof is correct.
-All actions on L1 require some gas to be spent. So, for example, if we aggregate some transactions into 1 block instead of 2, we will save some gas because we will need to send only one commitment and one proof instead of two. Thus, we aggregate ranges of L2 blocks into L1 batches and send one commitment/proof per L1 batch to save on gas costs.
+Accessing block numbers within zkSync API is similar to how you would do it on Ethereum. For example, `eth_blockNumber` returns the number 
+of the latest L2 block, and `eth_getBlockByNumber`, given a block number, returns the information about the requested block.
 
-### Ethereum Block Numbers Within zkSync
+For L1 batches, to retrieve the latest batch number, use zkSync API method `zks_L1BatchNumber`.
+Additionally, when querying for a given block information, you can find the batch number that includes the block. Within transaction receipts, the field `l1BatchNumber` is also the batch that includes the transaction. The field `l1BatchTxIndex` returns the transaction position among all of the batch transactions.
```

Comment:
> batch number that includes the block

while the meaning is clear, this sentence isn't accurate, as it is a _batch_ the includes the block, not the batch number